ouroboros/chainsync/compatibility/types.go

```diff
@@ -199,3 +199,40 @@ func (c *CompatibleValue) UnmarshalDynamoDBAttributeValue(item *dynamodb.Attribu
 	*c = CompatibleValue(s)
 	return nil
 }
+
+type CompatibleResult struct {
+	NextBlock        *CompatibleResultNextBlock
+	FindIntersection *CompatibleResultFindIntersection
+}
+
+func (c *CompatibleResult) UnmarshalJSON(data []byte) error {
+	var rfi CompatibleResultFindIntersection
```

Comment:
actually, slight bug; in the unmarshal method for CompatibleResultFindIntersection / CompatibleResultNextBlock, you have a line of code like
```
if err == nil && r.Tip != ""
```

or whatever... But you use the same property for both of them.

If you were to try to unmarshal a NextBlock as a FindIntersection here, it would succeed, because both of them have a Tip property.

You need to check for a unique property in each case: perhaps Direction and Intersection